internal/osbuildexecutor/runner-impl-aws-ec2.go

```diff
@@ -24,6 +30,174 @@ func prepareSources(manifest []byte, store string, extraEnv []string, result boo
+	client := http.Client{
+		Timeout: time.Second * 1,
+	}
+
+	for {
+		resp, err := client.Get(fmt.Sprintf("%s/api/v1/", host))
+		if err != nil {
+			logrus.Debugf("Waiting for secure instance continues: %v", err)
+		}
+		if resp != nil {
+			defer resp.Body.Close()
+			if resp.StatusCode == 200 {
+				return true
+			}
+			body, err := io.ReadAll(resp.Body)
+			if err != nil {
+				logrus.Warningf("Unable to read body waiting for secure instance: %v", err)
+			}
+			logrus.Debugf("Waiting for secure instance continues: %s", body)
+		}
+		select {
+		case <-ctx.Done():
+			logrus.Error("Timeout waiting for secure instance to spin up")
+			return false
+		default:
+			time.Sleep(time.Second)
+			continue
+		}
+	}
+}
+
+func writeInputArchive(cacheDir, store string, exports []string, manifestData []byte) (string, error) {
+	archive := filepath.Join(cacheDir, "input.tar")
+	control := filepath.Join(cacheDir, "control.json")
+	manifest := filepath.Join(cacheDir, "manifest.json")
+
+	controlData := struct {
+		Exports []string `json:"exports"`
+	}{
+		Exports: exports,
+	}
+	controlDataBytes, err := json.Marshal(controlData)
+	if err != nil {
+		return "", err
+	}
+	err = os.WriteFile(control, controlDataBytes, 0600)
+	if err != nil {
+		return "", err
+	}
+	err = os.WriteFile(manifest, manifestData, 0600)
+	if err != nil {
+		return "", err
+	}
+
+	cmd := exec.Command("tar",
+		"-C",
+		cacheDir,
+		"-cf",
+		archive,
+		filepath.Base(control),
+		filepath.Base(manifest),
+	)
+	if output, err := cmd.CombinedOutput(); err != nil {
+		logrus.Errorf("Unable to create input tar: %v\n%s", err, output)
+		return "", err
+	}
+	/* #nosec G204 */
+	cmd = exec.Command("tar",
+		"-C",
+		filepath.Dir(store),
+		"-rf",
+		archive,
+		filepath.Base(store),
+	)
+	if output, err := cmd.CombinedOutput(); err != nil {
+		logrus.Errorf("Unable to create input tar: %v\n%s", err, output)
+		return "", err
+	}
+
+	return archive, nil
+}
+
+func handleBuild(inputArchive, host string) (*osbuild.Result, error) {
+	client := http.Client{
+		Timeout: time.Minute * 60,
+	}
+	inputFile, err := os.Open(inputArchive)
+	if err != nil {
+		return nil, err
+	}
+	defer inputFile.Close()
+
+	resp, err := client.Post(fmt.Sprintf("%s/api/v1/build", host), "application/x-tar", inputFile)
+	if err != nil {
+		logrus.Errorf("Unable to request build from executor instance: %v", err)
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 201 {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			logrus.Warningf("Unable to read body waiting for build to run: %v", err)
+			return nil, err
+		}
+		return nil, fmt.Errorf("Something went wrong during executor build: %v\n%s", err, body)
+	}
+
+	var osbuildResult osbuild.Result
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("Something went wrong during executor build: %v\n%s", err, body)
+	}
+	err = json.Unmarshal(body, &osbuildResult)
```

Comment:
(super nitpick) we could use a `json.Decoder` which would avoid the io.ReadAll() as it would stream directly from the body into the json decoding (but fine to leave as is, this is a super nitpick). Unless I miss something of course, maybe there is a good reason to first get it all and then unmarshal :)